internal/installercache/installercache.go

```diff
@@ -1,61 +1,80 @@
 package installercache
 
 import (
+	"os"
 	"sync"
 
+	lru "github.com/hashicorp/golang-lru/v2"
 	"github.com/openshift/assisted-service/internal/oc"
 	"github.com/openshift/assisted-service/models"
 	"github.com/openshift/assisted-service/pkg/executer"
 	"github.com/openshift/assisted-service/pkg/mirrorregistries"
 	"github.com/sirupsen/logrus"
 )
 
-type installers struct {
-	sync.Mutex
-	releases map[string]*release
+//TODO: TwoQueueCache which support lfu policy does
+//      not support onEvict hook that we need in order
+//      to clear the files. Replace the cache when they
+//      do.
+type ReleaseCache struct {
+	releases *lru.Cache[string, *release]
+	log      logrus.FieldLogger
 }
 
 type release struct {
 	sync.Mutex
-	path string
+	Path string
 }
 
-var cache installers = installers{
-	releases: make(map[string]*release),
+func New(capacity int, log logrus.FieldLogger) *ReleaseCache {
+	rc := ReleaseCache{
+		log: log,
+	}
+	c, err := lru.NewWithEvict(capacity, rc.onEvicted)
+	if err != nil {
+		log.WithError(err).Errorln("Failed to initialize the release binaries cache")
```

Comment:
Why not ``Fatalln``? If we couldn't construct the cache I don't see a reason to continue running

Reply:
ok